### PR TITLE
begin using wallet-sdk to get account details

### PR DESCRIPTION
**Pull request by piyalbasu**
Breaking the Account view overhaul into smaller parts to make code review easier. This PR mostly focuses on using wallet-sdk instead of stellar-sdk to get the model for account data and subsequently updating components to ingest the new model

### CODE REVIEW

File: @shared/api/internal.ts [hunk 1]
```diff
@@ -183,32 +183,40 @@ export const confirmPassword = async (
...
 
   try {
-    response = await server.loadAccount(publicKey);
+    isFunded = await dataProvider.isAccountFunded();
   } catch (e) {
+    isFunded = false;
     console.error(e);
-    // TODO: Check that this is indeed a 404 before returning 0
-    return { balance: "0", isFunded: false };
   }
```
Comment (piyalbasu) [inline @shared/api/internal.ts:214]: using a separate try/catch here as I don't want an issue with `fetchAccountDetails` or `fetchPayments` to erroneously flag an account as "unfunded" and give someone a heart attack

File: extension/src/popup/components/account/AccountAssets.tsx [hunk 2]
```diff
@@ -34,18 +34,16 @@ const AssetTypeEl = styled.span`
...
-  <>
-    <AccountAssetsHeaderEl>Account assets</AccountAssetsHeaderEl>
-    <AssetWrapper>
-      <StellarLogoEl alt="Stellar logo" src={StellarLogo} />
-      <LumenBalanceEl>
-        {accountBalance} <AssetTypeEl>XLM</AssetTypeEl>
-      </LumenBalanceEl>
-    </AssetWrapper>
-  </>
+export const AccountAssets = ({ balances }: { balances: Balances }) => (
```
Comment (piyalbasu) [inline extension/src/popup/components/account/AccountAssets.tsx:37]: this view is still WIP - just swapping in the new data model

File: extension/src/popup/components/account/AccountDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  const handleDetailToggle = (isAssetsActive: boolean) => {
+    if (isAccountAssetsActive !== isAssetsActive) {
+      setIsAccountAssetsActive(isAssetsActive);
+    }
+  };
+
+  if (isFunded === null) {
+    return null;
+  }
```
Comment (piyalbasu) [inline extension/src/popup/components/account/AccountDetails.tsx:74]: again - just being pretty deliberate about now showing a "not funded" message unless we are *sure* it isn't funded

File: extension/src/popup/components/account/AccountHistory.tsx [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import React from "react";
```
Comment (piyalbasu) [inline extension/src/popup/components/account/AccountHistory.tsx:1]: placeholder view for account history

File: @shared/api/internal.ts [hunk 5]
```diff
@@ -183,32 +183,40 @@ export const confirmPassword = async (
...
+  let payments = null;
+  let balances = null;
+  let isFunded = null;
+
+  try {
+    ({ balances } = await dataProvider.fetchAccountDetails());
+    const transactionData = await dataProvider.fetchPayments({
+      limit: TRANSACTIONS_LIMIT,
+    });
+    payments = transactionData?.records || {};
```
Comment (quietbits) [inline @shared/api/internal.ts:204]: We could replace `{}` with `null` in this case if there is no need to know if it was "checked" already.

**Review by quietbits — APPROVED** [review]

**Pull request merged**